homeassistant/components/zwave_js/api.py

```diff
@@ -183,6 +243,56 @@ async def async_handle_failed_command_func(
     return async_handle_failed_command_func
 
 
+def get_provisioning_info(
+    msg: dict,
+) -> str | ProvisioningEntry | QRProvisioningInformation | None:
+    """Process provisioning parameters and return the appropriate value."""
+    if planned_provisioning_entry := msg.get(PLANNED_PROVISIIONING_ENTRY):
+        return ProvisioningEntry(
+            dsk=planned_provisioning_entry[DSK],
+            security_classes=[
+                SecurityClass(sec_cls)
+                for sec_cls in planned_provisioning_entry[SECURITY_CLASSES]
+            ],
+            additional_properties={
+                k: v
+                for k, v in planned_provisioning_entry.items()
```

Comment:
Are these items validated somehow?

Reply:
No, per the [interface `PlannedProvisioningEntry`](https://zwave-js.github.io/node-zwave-js/#/api/controller?id=provisionsmartstartnode) the only requirement is that the key is a string, but the key names and the value types are not defined. I am not sure how to add the string key check to the validation schema but I could do that check in the API command if you think it would be valuable

Reply:
The keys in JSON must be strings, so maybe we don't need to validate that?